chef/automate
# [A2-782] fetch real data for global projects filter

### :nut_and_bolt: Description
Swapped the fake data for real data 🎉 


### :+1: Definition of Done
- [x] options in global projects filter are populated by backend
- [x] data gets refreshed on login and periodically thereafter
- [x] global projects filter options are accurate for logged in user
- [x] Follow-up: project names display in the global projects filter - API currently only returns project IDs, but we want the names in the filter labels. we could get the names from the UI store BUT we can't be certain that a user with partial access to a project (i.e. can get `proj3` Teams) also has access to `proj3` as a resource itself ➡️[chore card](https://chefio.atlassian.net/browse/A2-832)


### :athletic_shoe: Demo Script / Repro Steps
set up your policies and some resources to play with:
```
export TOK=`chef-automate admin-token`
chef-automate iam upgrade-to-v2 --beta2.1

# (you can also create these in the UI, just make sure the IDs are the same)
curl -H "api-token: $TOK" -XPOST -d '{"id":"proj3","name":"project numero 3"}' https://localhost/apis/iam/v2beta/projects --insecure | jq
curl -H "api-token: $TOK" -XPOST -d '{"id":"proj4","name":"project numero 4"}' https://localhost/apis/iam/v2beta/projects --insecure | jq
curl -H "api-token: $TOK" -XPOST -d '{"id":"proj9","name":"project numero 9"}' https://localhost/apis/iam/v2beta/projects --insecure | jq

curl -kH "api-token: $TOK" -d '{"name": "t3", "id": "t3", "projects": ["proj3"]}'  https://a2-dev.test/apis/iam/v2beta/tokens | jq
curl -kH "api-token: $TOK" -d '{"name": "t4", "id": "t4", "projects": ["proj4"]}'  https://a2-dev.test/apis/iam/v2beta/tokens | jq

curl -kH "api-token: $TOK" -d '{"name": "t3", "id": "t3", "projects": ["proj3", "proj4"]}'  https://a2-dev.test/apis/iam/v2beta/teams | jq
curl -kH "api-token: $TOK" -d '{"name": "t4", "id": "t4", "projects": ["proj4"]}'  https://a2-dev.test/apis/iam/v2beta/teams | jq

curl -kH "api-token: $TOK" -d '{"name": "coolbean", "id": "coolbean", "password": "chefautomate"}'  https://a2-dev.test/apis/iam/v2beta/users | jq

# save in policy.json
{
  "id": "test",
  "name": "testing global project filter",
  "members": ["user:local:coolbean"],
  "projects": ["proj3", "proj4"],
  "statements": [
    {
      "effect": "ALLOW",
      "role": "iam-members-viewer",
      "projects": ["proj3", "proj4"]
    }
  ]
}
curl -kH "api-token: $TOK" -d @policy.json https://a2-dev.test/apis/iam/v2beta/policies | jq
```
2. start up the UI, log in as the admin, navigate to Settings, and try out the project filter. you should see all three projects you made
<img width="622" alt="Screen Shot 2019-05-02 at 2 54 33 PM" src="https://user-images.githubusercontent.com/21015366/57110257-bbc76780-6cec-11e9-9836-9fc5595da77c.png">

3. log in as `coolbean` with password `chef-automate`, navigate to Settings, and try out the project filter. you should only see `proje3` and `proj4`
<img width="613" alt="Screen Shot 2019-05-06 at 5 55 32 PM" src="https://user-images.githubusercontent.com/21015366/57264690-3adbd900-7028-11e9-8ea3-20976b785a59.png">

### :chains: Related Resources
https://github.com/chef/automate/pull/172

### :white_check_mark: Checklist
